### DISCUSSION TITLE
/cart/addItem does not update visible cart-Item-Count

### DISCUSSION THREAD

**Discussion by elaspix** [woocommerce/woocommerce-gutenberg-products-block#10705] [Q&A - Extension Integrations and Customizations] [answered]
I have a Javascript-component running in a html-Element within a default wooCommerce-Page that has the default navigation and basket-Item in the header of the page.

When the Javascript-component calls /cart/addItem using the storeAPI, I would like that the basket-item's counter updates so that the user can see that the product has reached the basket successfully. However, the basket keeps showing the "old" count and - using mouseover - is pretending that no product has been added. Only after a page reload the correct count is shown.

Is there a way to make wooCommerce update the shown count of products in the basket without a forced page reload and without changing the codebase of wooCommerce (e.g. adding filters etc.)

Best regards
Tobias

**Comment by opr** [comment]
Hi @elaspix, thanks for your question, I understand your cart isn't updating following a request to the store API.

I think this is expected behaviour since we wouldn't update any UI based on an API request alone, the code starting the request should use the data returned to update the UI, but with a little more information we may be able to give some pointers to figure this out anyway.

Could you confirm what you mean by:

>  default navigation and basket-Item 

What is the basket-Item you're referring to? Is it the Mini Cart Block? Could you maybe share a screenshot if possible?

**Reply by elaspix** [reply]
Hey opr, thanks for your answer, 

The basket-Item is located in the header top-right and shows a counter of products that are in the basket. This icon should update after a cart/addItem has been called (but it is not,  as it is shown in  "add_was_pressed_cartItemRed.jpg"). Only after a refresh of the page, the basket-item-count is updated (as it is shown in "after_manual_reload_productCounterRed.jpg"). 

So how do I make the basket-item-count show the updated number of products after I have added products to the basket using the storeAPI-call cart/addItem from my javascript-frontend?

I do not understand how the "code starting the request" can update the UI, I (my javascript code) has no access to the header-UI as it is the default wooCommerce header (see images). 

Thanks very much for your help!
Best regards
Tobias

![add_was_pressed_cartItemRed](https://github.com/woocommerce/woocommerce-blocks/assets/143009568/7fb37c0f-f231-4da1-92be-c450f57e5a58)
![after_manual_reload_productCounterRed](https://github.com/woocommerce/woocommerce-blocks/assets/143009568/29867eec-3c57-4d81-841f-c120136d9821)

**Comment by senadir** [comment]
As @opr said, you should either:

- Use the mini cart block and use data stores.
- Trigger a JS event `wc-blocks_added_to_cart` or a jQuery event `added_to_cart`.
- Manipulate DOM manually.

**Comment by elaspix** [comment]
@senadir 
Thanks for your response. I would have hoped to circumvent the "manipulate DOM manually"-Part as it is not very future-proof. It would be better to have API-Access to the cart to notify it for updating. 
Thanks a lot
Tobias

**Comment by senadir** [comment]
The events presented above are meant for that, you would trigger them and they would cause a legacy mini cart refresh.

Newer mini cart provide much more functionality.

**Comment by elaspix** [comment]
ah ok, thank you. But this means I must use the mini cart block and cannot keep the current default wooCommerce header with cart-icon (which has all I need already, except the updating problem)?

**Comment by senadir** [comment]
Using events is the only way to get this and keep the old cart yes.

**Comment by elaspix** [comment, marked as answer]
I have found that an event is registered to the body-element called: **updated_wc_div**
If this event is called after I have added items to the cart using StoreAPI's /cart/addItem, the cart-count updates as wished. 
I call this event via javascript using `jQuery("body").trigger("updated_wc_div")`
